coderd/provisionerdserver/provisionerdserver.go

```diff
@@ -1297,6 +1281,59 @@ func InsertWorkspaceResource(ctx context.Context, db database.Store, jobID uuid.
 			}
 		}
 
+		logSourceIDs := make([]uuid.UUID, 0, len(prAgent.Scripts))
+		logSourceCreatedAt := make([]time.Time, 0, len(prAgent.Scripts))
+		logSourceDisplayNames := make([]string, 0, len(prAgent.Scripts))
+		logSourceIcons := make([]string, 0, len(prAgent.Scripts))
+		scriptLogPaths := make([]string, 0, len(prAgent.Scripts))
+		scriptSources := make([]string, 0, len(prAgent.Scripts))
+		scriptCron := make([]string, 0, len(prAgent.Scripts))
+		scriptTimeout := make([]int32, 0, len(prAgent.Scripts))
+		scriptStartBlocksLogin := make([]bool, 0, len(prAgent.Scripts))
+		scriptRunOnStart := make([]bool, 0, len(prAgent.Scripts))
+		scriptRunOnStop := make([]bool, 0, len(prAgent.Scripts))
+
+		for _, script := range prAgent.Scripts {
+			logSourceIDs = append(logSourceIDs, uuid.New())
+			logSourceCreatedAt = append(logSourceCreatedAt, dbtime.Now())
```

Comment:
A pet-peeve of mine is using a var (`now := dbtime.Now(); for ... { ... }`), this way there's no confusion in the DB why script sources were created almost at the same time, but now quite.

Reply:
Agreed. I'll just change this to not use an array, because it really just needs the single val anyways.